### PR TITLE
Represent numpy ndarray as list in YAML

### PR DESCRIPTION
**Pull request by jsouter**
Also adds custom YAML dumper to represent Enum subclasses as values

### CODE REVIEW

**Comment by coretl** [comment]
Please can we have a test that shows the contents of a saved yaml file?

**Comment by coretl** [comment]
@rosesyrett @callumforrester tests are failing on main now because of hdf_streamer_det which we have deleted. Should we rebase this PR on top of add-detector-logic, or shall we just merge add-detector-logic and say we'll make subsequent PRs to fix issues?

**Comment by jsouter** [comment]
> Please can we have a test that shows the contents of a saved yaml file?

Yep, I'll put some in. For now the file being created, with default_flow_style set to None looks like
`- {child1.sig1: string}
- child2.sig1:
    VAL1: [1, 1, 1, 1, 1]
    VAL2: [1, 1, 1, 1, 1]
`

File: tests/core/_device/test_device_save_loader.py [hunk 1]
```diff
@@ -142,3 +148,48 @@ def save_my_device():
...
+
+    RE(save_my_device())
+
+    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
+        assert file.read() == """\
+- {child1.sig1: test_string}
+- child2.sig1:
+    VAL1: [1, 2, 3, 4, 5]
+    VAL2: [6, 7, 8, 9, 10]
+"""
```
Comment (coretl) [inline tests/core/_device/test_device_save_loader.py:174]: Black isn't happy, how about
```suggestion
        expected = """\
- {child1.sig1: test_string}
- child2.sig1:
    VAL1: [1, 2, 3, 4, 5]
    VAL2: [6, 7, 8, 9, 10]
"""
        assert file.read() == expected
```

**Review by coretl — COMMENTED** [review]
Please can you rebase on main then force push?

**Comment by rosesyrett** [comment]
I think it's best to rebase on top of add-detector-logic, and then merge it into add-detector-logic instead.

hopefully if we can get add-detector-logic checked for sanity (i.e. actually do a flyscan with it) we can then merge everything in, or merge add-detector-logic in and then merge this in. Either way, the important thing to me is we don't merge it until said sanity check has been done.

**Comment by coretl** [comment]
> I think it's best to rebase on top of add-detector-logic, and then merge it into add-detector-logic instead.

Good point, that's a much better idea

**Comment by jsouter** [comment]
> > I think it's best to rebase on top of add-detector-logic, and then merge it into add-detector-logic instead.
> 
> Good point, that's a much better idea

Okay, see this PR https://github.com/bluesky/ophyd-async/pull/31

**Comment by coretl** [comment]
Closed in favour of #36

**Pull request closed without merging**